addons/refuel/functions/fnc_refuel.sqf

```diff
@@ -46,32 +44,39 @@ private _maxFuel = getNumber (configFile >> "CfgVehicles" >> (typeOf _target) >>
         _nozzle setVariable [QGVAR(isConnected), false, true];
         _nozzle setVariable [QGVAR(sink), objNull, true];
         _sink setVariable [QGVAR(nozzle), objNull, true];
-        [_pfID] call cba_fnc_removePerFrameHandler;
+        [_pfID] call CBA_fnc_removePerFrameHandler;
     };
 
     private _finished = false;
     private _fueling = _nozzle getVariable [QGVAR(isRefueling), false];
     if (_fueling) then {
         if (isEngineOn _sink) exitWith {
+            _nozzle setVariable [QGVAR(lastTickMissionTime), nil, true];
             _nozzle setVariable [QGVAR(isRefueling), false, true];
         };
         private _fuelInSource = [_source] call FUNC(getFuel);
         if (_fuelInSource == 0) exitWith {
             [LSTRING(Hint_SourceEmpty), 2, _unit] call EFUNC(common,displayTextStructured);
+            _nozzle setVariable [QGVAR(lastTickMissionTime), nil, true];
             _nozzle setVariable [QGVAR(isRefueling), false, true];
         };
+
+        // Calculate rate using mission time to take time acceleration and pause into account
+        private _rateTime = _rate * (CBA_missionTime - (_nozzle getVariable [QGVAR(lastTickMissionTime), CBA_missionTime]));
+        _nozzle setVariable [QGVAR(lastTickMissionTime), CBA_missionTime, true];
```

Comment:
I think that setting this as public is overkill. If this is only set locally and locality changes, all that'd happen is that for a single tick the _rateTime would be zero, but it'd go back to normal on the next tick; hardly an issue.

The whole function is a bit public-heavy; but that's probably a matter for a different PR.